src/main/java/org/tugraz/sysds/runtime/util/DataConverter.java

```diff
@@ -758,7 +759,28 @@ public static TensorBlock convertToTensorBlock(MatrixBlock mb, ValueType vt) {
 		return ret;
 	}
 
-	public static MatrixBlock[] convertToMatrixBlockPartitions( MatrixBlock mb, boolean colwise ) 
+	public static HeterogTensor convertToHeterogTensor(MatrixBlock mb, ValueType vt) {
```

Comment:
Maybe change the signature to return a TensorBlock, and pass a boolean argument if this should be a BasicTensorBlock or DataTensorBlock